tasks/generate_ca.yml

```diff
@@ -0,0 +1,68 @@
+---
+- name: Create CA key directory
+  file:
+    path: "{{ certificates_ca_key_dir }}"
+    state: directory
+  delegate_to: "{{ certificates_ca_server }}"
+
+- name: Generate CA privatekey
+  openssl_privatekey:
+    path: "{{ certificates_ca_key_dir }}/ca-key.pem"
+    size: "{{ certificates_key_length }}"
+  delegate_to: "{{ certificates_ca_server }}"
+
+- name: Generate CA signing request
+  openssl_csr:
+    path: "{{ certificates_ca_key_dir }}/ca.csr"
+    privatekey_path: "{{ certificates_ca_key_dir }}/ca-key.pem"
+    common_name: "{{ certificates_ca_common_name }}"
+    country_name: "{{ certificates_ca_country_name }}"
+    organization_name: "{{ certificates_ca_organization_name }}"
+    email_address: "{{ certificates_ca_email_address }}"
+  delegate_to: "{{ certificates_ca_server }}"
+
+- name: Generate CA certificate
+  openssl_certificate:
+    path: "{{ certificates_ca_key_dir }}/ca.crt"
+    privatekey_path: "{{ certificates_ca_key_dir }}/ca-key.pem"
+    csr_path: "{{ certificates_ca_key_dir }}/ca.csr"
+    provider: selfsigned
+  delegate_to: "{{ certificates_ca_server }}"
+
+- name: Install ca.conf config file
+  template:
+    src: ca.conf.j2
+    dest: "{{ certificates_ca_key_dir }}/ca.conf"
+    owner: root
+    group: root
+    mode: 744
+  delegate_to: "{{ certificates_ca_server }}"
+
+- name: Create initial certificate revocation list squence number
+  shell: "echo 00 > crl_number"
+  args:
+    chdir: "{{certificates_ca_key_dir}}"
+    creates: crl_number
+  delegate_to: "{{ certificates_ca_server }}"
+
+- name: Install revocation script
+  template:
+    src: revoke.sh.j2
+    dest: "{{certificates_ca_key_dir}}/revoke.sh"
+    owner: root
+    group: root
+    mode: 744
```

Comment:
Why not 755? Also you forgot the 0 at the beginning

Reply:
fixed